packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/custom_onsite_payments_app_extension_schema.ts

```diff
@@ -0,0 +1,67 @@
+import {BaseSchema} from '../../schemas.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+export type CustomOnsitePaymentsAppExtensionConfigType = zod.infer<typeof CustomOnsitePaymentsAppExtensionSchema>
+
+const MAX_LABEL_SIZE = 50
+
+export const CustomOnsitePaymentsAppExtensionSchema = BaseSchema.extend({
+  targeting: zod.array(zod.object({target: zod.literal('payments.custom-onsite.render')})).length(1),
```

Comment:
I don't think we can extract this field to a base schema, since there's a link between the target type and the expected fields it can accept that specific type. If we were to extract the targeting's target property as a union, validations for schemas would not match property. How did you think about extracting this property?